crypto/rand/rand_lib.c

```diff
@@ -373,13 +378,18 @@ RAND_POOL *rand_pool_new(int entropy_requested, size_t min_len, size_t max_len)
     pool->max_len = (max_len > RAND_POOL_MAX_LENGTH) ?
         RAND_POOL_MAX_LENGTH : max_len;
 
-    pool->buffer = OPENSSL_secure_zalloc(pool->max_len);
+    if (secure)
+        pool->buffer = OPENSSL_secure_zalloc(pool->max_len);
+    else
+        pool->buffer = OPENSSL_zalloc(pool->max_len);
```

Comment:
of course I could change that to OPENSSL_malloc and add a memset
guarded by `#ifdef __SANITIZE_MEMORY__` maybe with a short comment,
would that be worth a try?

Reply:
I don't know... feels like an ugly hack. Is really worth the performance gain? In view of @paulidale's #9428, I'd say no.

Reply:
Yeah, okay.